### PR TITLE
Translate introduction to russian

### PR DESCRIPTION
**Pull request by eupn**
Just leveraged knowledge of my native language to make Grin and MW more accessible for people from the largest country in the world™.

### CODE REVIEW

File: doc/intro.ru.md [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+* Масштабироваться в основном с количеством пользователей
+и минимально с количеством транзакций. Это приводит к значительной
+экономии дискового пространства, по сравнению с другими блокчейнами.
+* Сильная и доказанная криптография. МимблВимбл опирается на проверенные
+временем Эллиптические Кривые, используемые и тестируемые десятилетиями.
+* Простой дизайн системы упрощает аудит и дальнейшую поддержку.
+* Ведётся сообществом, использует ASIC-устойчивый алгоритм майнинга
+(Cuckoo Cycle), приветствуя децентрализацию майнеров.
+
+### Завязывание Языка для Чайников
```
Comment (hashmap) [inline doc/intro.ru.md]: Not sure about `Чайников` however `Всех`also sounds a bit different

File: doc/intro.ru.md [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+тогда мы также можем вычислить `(k+j)*H`, что равняется `k*H + j*H`.
+Сложение и скалярное умножение на Эллиптической Кривой удовлетворяет свойствам коммутативности и
+ассоциативности сложения и умножения:
+
+    (k+j)*H = k*H + j*H
+
+В Эллиптической Криптографии, если мы выберем большое значение _k_ как публичный ключ,
+тогда произведение `k*H` станет соответствующим публичным ключом.
+Даже если кто-то знает значение публичного ключа `k*H`, вычисление _k_ близко к невозможному
+(другими словами, не смотря на тривиальность умножения, деление точек Эллиптической Кривой является
```
Comment (hashmap) [inline doc/intro.ru.md]: I think `несмотря`?

File: doc/intro.ru.md [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+Эти точки можно складывать, вычитать или умножать на целые числа (так же называемые скалярами).
+Пусть _k_ является целым числом, тогда, используя скалярное умножение, мы можем вычислить
+`k*H`, что так же является точкой на кривой _C_. Пусть дано другое целое число _j_,
+тогда мы также можем вычислить `(k+j)*H`, что равняется `k*H + j*H`.
+Сложение и скалярное умножение на Эллиптической Кривой удовлетворяет свойствам коммутативности и
+ассоциативности сложения и умножения:
+
+    (k+j)*H = k*H + j*H
+
+В Эллиптической Криптографии, если мы выберем большое значение _k_ как публичный ключ,
```
Comment (hashmap) [inline doc/intro.ru.md]: I think `открытый/закрытый ключ` is more or less official translation of `public/private key` https://ru.wikipedia.org/wiki/%D0%9A%D1%80%D0%B8%D0%BF%D1%82%D0%BE%D1%81%D0%B8%D1%81%D1%82%D0%B5%D0%BC%D0%B0_%D1%81_%D0%BE%D1%82%D0%BA%D1%80%D1%8B%D1%82%D1%8B%D0%BC_%D0%BA%D0%BB%D1%8E%D1%87%D0%BE%D0%BC

File: doc/intro.ru.md [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+могут быть совершены без раскрытия количеств монет, передаваемых этими транзакциями.
+
+Примечательно, что эта идея была выведена из 
+[Конфиденциальных Транзакций](https://www.elementsproject.org/elements/confidential-transactions/) Грега Максвелла,
+которые, в свою очередь, сами основаны на предложении Адама Бэка для гомоморфных значений, применимых к
+Биткоину.
+
+#### Владение
+
+Выше мы ввели приватный ключ в качестве фактора сокрытия, чтобы засекретить информацию о
```
Comment (hashmap) [inline doc/intro.ru.md]: `сокрыть информацию`? later on (line 165) `раскрыто` is used, so it would sound more symmetric

File: doc/intro.ru.md [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+Например, она выбрала 133, тогда в блокчейн будет записано:
+
+    Y - Xi = (113*G + 3*H) - (28*G + 3*H) = 85*G + 0*H
+
+Эта сумма (транзакция) больше не сводится к нулю и мы имеем _избыточное_ значение на _G_ (85), 
+которое является результатом сложения всех факторов сокрытия. Но из-за того, что
+произведение `85*G` будет являться корректным публичным ключом на ЭК _G_ с приватным ключом 85,
+для любого x и y, только если `y = 0`, сумма `x*G + y*H` будет являться публичным ключом на _G_.
+
+Всё что нужно, это проверить, что (`Y - Xi`) - валидный публичный ключ на кривой _G_ и
```
Comment (hashmap) [inline doc/intro.ru.md]: Валидный  - not sure how to replace it, perhaps действительно открытый ключ?

File: doc/intro.ru.md (hunk 1 above)
Comment (eupn) [reply doc/intro.ru.md]: I understood "for everyone" as "easy to understand for everyone" and "для чайников" is idiomatic way to say "easy to understand" or "in simple words".

File: doc/intro.ru.md (hunk 2 above)
Comment (eupn) [reply doc/intro.ru.md]: Nice catch! Can you please suggest an edition via new github suggestions feature?

File: doc/intro.ru.md (hunk 3 above)
Comment (eupn) [reply doc/intro.ru.md]: Yes, turns out it is more widespread wording. Will replace that.

File: doc/intro.ru.md (hunk 5 above)
Comment (eupn) [reply doc/intro.ru.md]: Валидный means valid, but your suggestion is good.

File: doc/intro.ru.md (hunk 4 above)
Comment (eupn) [reply doc/intro.ru.md]: 👌

File: doc/intro.ru.md (hunk 1 above)
Comment (hashmap) [reply doc/intro.ru.md]: I get it, however "для чайников" is a translation of "For Dummies", well known series of books, which has some different connotations. I think the direct translation Для Всех sounds better

Comment (eupn) [reply doc/intro.ru.md]: Got it. 👍

File: doc/intro.ru.md (hunk 2 above)
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
(другими словами, несмотря на тривиальность умножения, деление точек Эллиптической Кривой является
```

File: doc/intro.ru.md [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+Пусть _k_ является целым числом, тогда, используя скалярное умножение, мы можем вычислить
+`k*H`, что так же является точкой на кривой _C_. Пусть дано другое целое число _j_,
+тогда мы также можем вычислить `(k+j)*H`, что равняется `k*H + j*H`.
+Сложение и скалярное умножение на Эллиптической Кривой удовлетворяет свойствам коммутативности и
+ассоциативности сложения и умножения:
+
+    (k+j)*H = k*H + j*H
+
+В Эллиптической Криптографии, если мы выберем большое значение _k_ как открытый ключ,
+тогда произведение `k*H` станет соответствующим публичным ключом.
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
тогда произведение `k*H` станет соответствующим открытым ключом.
```

File: doc/intro.ru.md [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+Даже если кто-то знает значение публичного ключа `k*H`, вычисление _k_ близко к невозможному
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
Даже если кто-то знает значение открытого ключа `k*H`, вычисление _k_ близко к невозможному
```

File: doc/intro.ru.md [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+(другими словами, несмотря на тривиальность умножения, деление точек Эллиптической Кривой является
+крайне сложным). 
+
+Предыдущая формула `(k+j)*H = k*H + j*H`, где _k_ и _j_ хранятся в тайне, 
+показывает, что открытый ключ может быть получен путём сложения двух приватных ключей
+и является идентичным сложению двух соответствующих публичных ключей. Например, в Биткоин
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
и является идентичным сложению двух соответствующих открытых ключей. Например, в Биткоин
```

File: doc/intro.ru.md [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+такое же, как и у вас, чтобы свести сумму в ноль) и тогда вы можете украсть деньги у Кэрол назад!
+
+Для решения этой проблемы, Кэрол использует закрытый ключ, который выбрала сама.
+Например, она выбрала 133, тогда в блокчейн будет записано:
+
+    Y - Xi = (113*G + 3*H) - (28*G + 3*H) = 85*G + 0*H
+
+Эта сумма (транзакция) больше не сводится к нулю и мы имеем _избыточное_ значение на _G_ (85), 
+которое является результатом сложения всех факторов сокрытия. Но из-за того, что
+произведение `85*G` будет являться корректным публичным ключом на ЭК _G_ с приватным ключом 85,
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
произведение `85*G` будет являться корректным открытым ключом на ЭК _G_ с приватным ключом 85,
```

File: doc/intro.ru.md [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+для любого x и y, только если `y = 0`, сумма `x*G + y*H` будет являться публичным ключом на _G_.
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
для любого x и y, только если `y = 0`, сумма `x*G + y*H` будет являться открытым ключом на _G_.
```

File: doc/intro.ru.md [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+Напомним, что транзакция состоит из:
+
+* Набора входов, которые тратят набор предыдущих выходов.
+* Набора новых выходов (Обязательств Педерсена).
+* Ядра транзакции, состоящего из:
+  * Избыточного значения (Обязательство Педерсена для нуля).
+  * Подпись транзакции, использующая избыточное значение как ключ.
+
+Транзакция подписывается и подпись включается в _ядро транзакции_. Подпись генерируется с использованием _избытка ядра_
+в качестве публичного ключа, доказывая, что транзакция сводится к нулю.
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
в качестве открытого ключа, доказывая, что транзакция сводится к нулю.
```

File: doc/intro.ru.md [hunk 12]
```diff
@@ -0,0 +1,432 @@
...
+    (k+j)*H = k*H + j*H
+
+В Эллиптической Криптографии, если мы выберем большое значение _k_ как открытый ключ,
+тогда произведение `k*H` станет соответствующим публичным ключом.
+Даже если кто-то знает значение публичного ключа `k*H`, вычисление _k_ близко к невозможному
+(другими словами, несмотря на тривиальность умножения, деление точек Эллиптической Кривой является
+крайне сложным). 
+
+Предыдущая формула `(k+j)*H = k*H + j*H`, где _k_ и _j_ хранятся в тайне, 
+показывает, что открытый ключ может быть получен путём сложения двух приватных ключей
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
показывает, что открытый ключ может быть получен путём сложения двух закрытых ключей
```

File: doc/intro.ru.md [hunk 13]
```diff
@@ -0,0 +1,432 @@
...
+### Создание Транзакций в МимблВимбл
+
+Структура транзакций указывает на ключевые принципы МимблВимбла: нерушимую
+гарантию приватности и конфиденциальности.
+
+Проверка транзакций МимблВимбла опирается на два основных свойства:
+
+* **Проверка нулевых сумм.** Сумма выходов минус сумма входов всегда равняется нулю,
+  это доказывает, что транзакция не создаёт новых монет, при этом _без раскрытия реальных сумм переводов_.
+* **Владение приватным ключом.** Как и у большинства других криптовалют, владение
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
* **Владение закрытым ключом.** Как и у большинства других криптовалют, владение
```

File: doc/intro.ru.md [hunk 14]
```diff
@@ -0,0 +1,432 @@
...
+  выходами транзакции гарантируется владением приватного ключа. Однако доказательство того,
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
  выходами транзакции гарантируется владением закрытого ключа. Однако доказательство того,
```

File: doc/intro.ru.md [hunk 15]
```diff
@@ -0,0 +1,432 @@
...
+  что некто владеет приватным ключом, достигается иначе, нежели простой подписью транзакции.
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
  что некто владеет закрытым ключом, достигается иначе, нежели простой подписью транзакции.
```

File: doc/intro.ru.md [hunk 16]
```diff
@@ -0,0 +1,432 @@
...
+тогда (без учёта комиссий):
+
+* vi1 и vi2 это входы.
+* vo3 это выход.
+
+Такие, что:
+
+    vi1 + vi2 = vo3
+
+Создав приватные ключи как факторы сокрытия для каждого из значений и заменив их на
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
Создав закрытые ключи как факторы сокрытия для каждого из значений и заменив их на
```

File: doc/intro.ru.md [hunk 17]
```diff
@@ -0,0 +1,432 @@
...
+Биткоину.
+
+#### Владение
+
+Выше мы ввели закрытый ключ в качестве фактора сокрытия, чтобы сокрыть информацию о 
+количестве передаваемых транзакцией монет. Вторая идея, которую предоставляет МимблВимбл, это 
+то, что этот же самый ключ может использоваться для доказательства владения монетами.
+
+Алиса отправляет вам 3 монеты и, чтобы сокрыть количество, вы выбрали 28 как ваш
+фактор сокрытия (заметим, что на практике фактор сокрытия, будучи приватным ключом,
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
фактор сокрытия (заметим, что на практике фактор сокрытия, будучи закрытым ключом, 
```

File: doc/intro.ru.md [hunk 18]
```diff
@@ -0,0 +1,432 @@
...
+Сумма _X_ является видимой для всех, а значение 3 известно только вам и Алисе. Ну а число 28 известно только вам.
+
+Чтобы передать эти 3 монеты снова, протокол требует, чтобы число 28 было каким-то образом раскрыто.
+Чтобы показать принцип работы, допустим, что вы хотите передать те же 3 монеты Кэрол.
+Тогда вам нужно создать простую транзакцию, такую, что:
+
+    Xi => Y
+
+Где _Xi_ это выход, который тратит ваш вход _X_, а Y это выход для Кэрол. Не существует способа создать
+такую транзакцию без знания вашего приватного ключа 28. Разумеется, если Кэрол решит принять монеты из этой
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
такую транзакцию без знания вашего закрытого ключа 28. Разумеется, если Кэрол решит принять монеты из этой
```

File: doc/intro.ru.md [hunk 19]
```diff
@@ -0,0 +1,432 @@
...
+транзакции, ей нужно будет узнать как значение вашего приватного ключа, так и значение, которое этой
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
транзакции, ей нужно будет узнать как значение вашего закрытого ключа, так и значение, которое этой
```

File: doc/intro.ru.md [hunk 20]
```diff
@@ -0,0 +1,432 @@
...
+транзакцией переводится. Таким образом:
+
+    Y - Xi = (28*G + 3*H) - (28*G + 3*H) = 0*G + 0*H
+
+Проверяя, что всё свелось к нулю, мы в очередной раз убедимся, что новых монет создано не было.
+
+Но постойте-ка! Теперь вы знаете значение приватного ключа из выхода Кэрол (которое, в этом случае, должно быть
```
Comment (hashmap) [inline doc/intro.ru.md]:
```suggestion
Но постойте-ка! Теперь вы знаете значение закрытого ключа из выхода Кэрол (которое, в этом случае, должно быть
```

**Pull request merged**